twinify/napsu_mq/maximum_entropy_model.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2022 twinify Developers and their Assignees
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Union
+
+import numpyro
+import numpyro.distributions as numdist
+import jax.numpy as jnp
+from twinify.napsu_mq.markov_network_jax import MarkovNetworkJax
+
+
+def normal_prior_model_numpyro(dp_suff_stat: jnp.ndarray, n: int, sigma_DP: float, prior_mu: Union[float, jnp.ndarray],
+                               prior_sigma: float, med: MarkovNetworkJax):
+    """NumPyro model for NAPSU-MQ with isotropic Gaussian prior.
+    Args:
+        dp_suff_stat (jax.ndarray): Noisy sufficient statistic.
+        n (int): Number of datapoints.
+        sigma_DP (float): Noise standard deviation.
+        prior_mu (float or jax.ndarray): Prior mean.
+        prior_sigma (float): Prior standard deviation.
+        med (MarkovNetworkJax): An implementation of MED.
+    """
+    d = med.suff_stat_d
+    lambda_d = med.lambda_d
+
+    lambdas = numpyro.sample("lambdas", numdist.MultivariateNormal(jnp.ones(lambda_d) * prior_mu,
+                                                                   covariance_matrix=jnp.eye(
+                                                                       lambda_d) * prior_sigma ** 2))
+    mean, cov = med.suff_stat_mean_and_cov(lambdas)
+    return numpyro.sample("a_hat",
```

Comment:
We'd need to somehow get the privacy noise from the d3p.random PRNG, but I have not clue how to do this right now inside of a numpyro model.. so maybe for now we can live with this but we need to find a way to do this in the future

Reply:
I don't think that's necessary here. The DP noise is not sampled here, and the randomness in the MCMC algorithms doesn't need to be cryptographically secure.

Reply:
Ah, alright; I have been misreading this. Thanks.